src/js_native_api_v8.cc

```diff
@@ -1335,10 +1335,17 @@ napi_status napi_create_string_utf8(napi_env env,
   CHECK_ENV(env);
   CHECK_ARG(env, result);
 
-  v8::Local<v8::String> s;
-  CHECK_NEW_FROM_UTF8_LEN(env, s, str, length);
-
-  *result = v8impl::JsValueFromV8LocalValue(s);
+  RETURN_STATUS_IF_FALSE(env,
+      (length == NAPI_AUTO_LENGTH) || length <= INT_MAX,
+      napi_invalid_arg);
```

Comment:
Actually, I’m wondering if the other string creation methods should also have this? I think it’s missing from them and may cause problems when `length` overflows.

Reply:
Yeah, I was wondering about that. Did you want me to add it to this pull request? Presumably tests would need to be added as well -- but there I'm not sure what ought to be done!

Reply:
> Did you want me to add it to this pull request?

I think it would make sense given that this touches the same code and would otherwise lead to merge conflicts. (It can be a different commit or a different PR, and it’s nothing you *have* to do in order for this to land, imo :smile:)

> Presumably tests would need to be added as well -- but there I'm not sure what ought to be done!

That would be amazing (`test/js-native-api/test_string` would be the place to look), but I don’t think we’re currently testing this for utf-8, so I think it should be okay to not add tests here.

Reply:
Probably best to create another issue for that?

Reply:
If we don't add tests as part of this PR, it would be good to add an issue describing the tests that should be added.  It will help ensure we don't forget about them and might also be a would task for somebody to take on if they are interested.

Reply:
Not sure if it was clear, but I did add the tests. I think this is now resolved.